src/porepy/numerics/fv/biot.py

```diff
@@ -365,48 +370,13 @@ def _discretize_mech(self, g, data):
         num_subhfno = subcell_topology.subhfno.size
 
         num_nodes = np.diff(g.face_nodes.indptr)
-        sgn = g.cell_faces[subcell_topology.fno, subcell_topology.cno].A
-
-        # The pressure gradient term in the mechanics equation is discretized
-        # as a force on the faces. The right hand side is thus formed of the
-        # normal vectors.
-        def build_rhs_normals_single_dimension(dim):
-            val = (
-                g.face_normals[dim, subcell_topology.fno]
-                * sgn
-                / num_nodes[subcell_topology.fno]
-            )
-            mat = sps.coo_matrix(
-                (val.squeeze(), (subcell_topology.subfno, subcell_topology.cno)),
-                shape=(subcell_topology.num_subfno, subcell_topology.num_cno),
-            )
-            return mat
-
-        rhs_normals = build_rhs_normals_single_dimension(0)
-        for iter1 in range(1, nd):
-            this_dim = build_rhs_normals_single_dimension(iter1)
-            rhs_normals = sps.vstack([rhs_normals, this_dim])
-
-        rhs_normals = bound_exclusion_mech.exclude_dirichlet(rhs_normals)
-
-        num_dir_subface = (
-            bound_exclusion_mech.exclude_neu.shape[1]
-            - bound_exclusion_mech.exclude_neu.shape[0]
-        )
-        # No right hand side for cell displacement equations.
-        rhs_normals_displ_var = sps.coo_matrix(
-            (
-                nd * subcell_topology.num_subfno - num_dir_subface,
-                subcell_topology.num_cno,
-            )
-        )
 
-        # Why minus?
-        rhs_normals = -sps.vstack([rhs_normals, rhs_normals_displ_var])
-        del rhs_normals_displ_var
+        num_subfno_unique = subcell_topology.num_subfno_unique
+        num_subfno = subcell_topology.num_subfno
+        num_cno = subcell_topology.num_cno
 
         # Call core part of MPSA
```

Comment:
Will this tacitly also construct the mpsa discretization matrices, and if so, do we risk overwriting values? 

@IvarStefansson, @rbe051 : We need to be careful here, at the very least, we should do a proper documentation on the relation between the components that together form mpsa for biot.

Reply:
Yes, but we don't (possible) overwrite before line 458. And you only overwrite if you use the same keyword. With the new parameter/keyword-structure I think it is fair to leave this to the user

Reply:
We might want to revisit the idea of separate keywords for input (parameters) and output (discretization_matrices)